common/admin.py

```diff
@@ -4,10 +4,40 @@
 from django.contrib.auth.admin import UserAdmin
 
 
+class BaseByTeacherFilter(admin.SimpleListFilter):
+    """
+    Base class to filter by Teacher.
+
+    From: https://books.agiliq.com/projects/django-admin-cookbook/en/latest/filtering_calculated_fields.html
+    and https://stackoverflow.com/questions/880684/in-django-admin-how-to-filter-users-by-group
+    """
+    title = 'teacher'
+    parameter_name = 'teacher'
+
+    def lookups(self, request, model_admin):
+        teachers = User.objects.filter(groups__name='teachers')
+        items = ( (t.id, t.username) for t in teachers )
+
+        return tuple(items)
```

Comment:
`items` is already a tuple, so it can be returned immediately

Reply:
Well, `items` is a generator and I didn't know it'd work. I can definitely test it.